medusa/search/core.py

```diff
@@ -792,6 +778,7 @@ def collect_candidates(found_results, provider, multi_results, single_results, s
                   if result in (SEASON_RESULT, MULTI_EP_RESULT))
     candidates = list(itertools.chain(*candidates))
     if candidates:
+        # FIXME: collect_multi_candidates can potentially result in downloading multiple multi-ep results.
```

Comment:
This needs to be handled in collect_results as we need all single episode results and all multi episode results to determine the needed episode results.

Reply:
And it should be already working like that.

Reply:
Yes, because the filtering is done in combine_results as we need the single episode results.

Reply:
I'm saying that for example, nyaa returns one release that's multi_ep. But after this method, it's duplicated? Why?